### PR TITLE
Add sharp config to image transformer

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

Add a sharp config file to set some defaults for compressing images

Based on #6624

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.00s | -11.00ms |
| Cached | 295.00ms | +8.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 80.00ms | -552.00ms 🚀 |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 80.00ms | -554.00ms 🚀 |
| dist/modern/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 80.00ms | -554.00ms 🚀 |
| dist/legacy/index.66850ffd.js | 1.59kb | +0.00b | 865.00ms | -66.00ms 🚀 |
| dist/legacy/index.01f0ce7e.js | 1.20kb | +0.00b | 864.00ms | -67.00ms 🚀 |
| dist/modern/index.65a4b415.js | 1.13kb | +0.00b | 865.00ms | -66.00ms 🚀 |
| dist/legacy/index.html | 826.00b | +0.00b | 1.02s | -65.00ms 🚀 |
| dist/modern/index.html | 749.00b | +0.00b | 1.02s | -66.00ms 🚀 |
| dist/legacy/index.5ba75694.css | 94.00b | +0.00b | 1.14s | -58.00ms 🚀 |
| dist/modern/index.14235cd8.css | 94.00b | +0.00b | 1.14s | -57.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 10.36s | -267.00ms |
| Cached | 465.00ms | -25.00ms 🚀 |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.09m | +839.00ms |
| Cached | 1.69s | +80.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.91149f3d.js | 1.78mb | -2.00b 🚀 | 18.93s | +93.00ms |
| dist/editorView.348a3df8.js | 594.92kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/popup.2063521e.js | 209.67kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/Toolbar.d426c7e1.js | 107.23kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/Modal.24a4c4b4.js | 45.31kb | +15.00b ⚠️ | 49.49s | +12.92s ⚠️ |
| dist/js.0f0bb621.js | 17.25kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/ui.0572f571.js | 14.94kb | +0.00b | 49.49s | +12.91s ⚠️ |
| dist/smartMediaEditor.9b4e37fb.js | 13.25kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/dropzone.0d183115.js | 12.15kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/EmojiPickerComponent.55e0a64c.js | 3.73kb | +0.00b | 49.49s | +12.91s ⚠️ |
| dist/png-chunks-extract.94b5b9fc.js | 3.58kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/dropzone.cf53fd41.js | 3.29kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/clipboard.969d5ad7.js | 2.93kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/ResourcedEmojiComponent.78f6ddfc.js | 2.12kb | +0.00b | 49.49s | +12.91s ⚠️ |
| dist/browser.09200cd5.js | 1.69kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/media-card-analytics-error-boundary.75fac461.js | 1.12kb | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/media-picker-analytics-error-boundary.1602c912.js | 966.00b | +0.00b | 49.49s | +12.92s ⚠️ |
| dist/simpleHasher.2a6e12b6.js | 643.00b | +0.00b | 49.49s | +12.91s ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/pdfRenderer.8a77c159.js | 1.11mb | +0.00b | 51.59s | -2.63s 🚀 |
| dist/index.f18013f4.js | 694.71kb | -15.00b 🚀 | 51.56s | -2.63s 🚀 |
| dist/editorView.bc3f0b86.js | 594.92kb | +0.00b | 50.59s | -3.03s 🚀 |
| dist/popup.2439c3e2.js | 209.67kb | +0.00b | 50.59s | -3.03s 🚀 |
| dist/index.e3bfd549.js | 145.75kb | +0.00b | 51.56s | -2.64s 🚀 |
| dist/Toolbar.4a4ac65b.js | 107.23kb | +0.00b | 50.59s | -3.03s 🚀 |
| dist/media-viewer.3775e851.js | 74.93kb | +0.00b | 51.59s | -2.63s 🚀 |
| dist/card.2dc6e26b.js | 62.67kb | +0.00b | 51.55s | -2.64s 🚀 |
| dist/card.684564f7.js | 60.71kb | +0.00b | 51.59s | -2.64s 🚀 |
| dist/Modal.1243616f.js | 45.31kb | +15.00b ⚠️ | 50.59s | -3.03s 🚀 |
| dist/ui.87cf5598.js | 14.94kb | +0.00b | 50.60s | -3.02s 🚀 |
| dist/smartMediaEditor.a53fadfb.js | 13.25kb | +0.00b | 50.59s | -3.03s 🚀 |
| dist/dropzone.0d183115.js | 12.15kb | +0.00b | 50.59s | -3.03s 🚀 |
| dist/workerHasher.07001132.js | 11.83kb | +0.00b | 37.61s | -2.38s 🚀 |
| dist/index.c71e39bf.js | 6.18kb | +0.00b | 51.59s | -2.64s 🚀 |
| dist/card.45f850d6.js | 5.77kb | +0.00b | 51.55s | -2.64s 🚀 |
| dist/media-viewer.141c77cc.js | 4.50kb | +0.00b | 37.60s | -2.38s 🚀 |
| dist/media-viewer.22c1906b.js | 3.93kb | +0.00b | 37.60s | -2.38s 🚀 |
| dist/EmojiPickerComponent.55e0a64c.js | 3.73kb | +0.00b | 50.60s | -3.02s 🚀 |
| dist/index.57b81618.css | 3.47kb | +0.00b | 51.83s | -2.66s 🚀 |
| dist/dropzone.a10cce43.js | 3.29kb | +0.00b | 50.59s | -3.03s 🚀 |
| dist/Modal.7f4c60b3.js | 3.15kb | +0.00b | 51.55s | -2.64s 🚀 |
| dist/clipboard.f917539f.js | 2.93kb | +0.00b | 50.59s | -3.03s 🚀 |
| dist/ResourcedEmojiComponent.78f6ddfc.js | 2.12kb | +0.00b | 50.60s | -3.02s 🚀 |
| dist/card.6492331a.js | 2.04kb | +0.00b | 37.60s | -2.38s 🚀 |
| dist/browser.09200cd5.js | 1.69kb | +0.00b | 50.59s | -3.03s 🚀 |
| dist/workerHasher.ca25f4e9.js | 1.63kb | +0.00b | 51.59s | -2.63s 🚀 |
| dist/Modal.d35ab6da.js | 1.16kb | +0.00b | 37.61s | -2.38s 🚀 |
| dist/media-card-analytics-error-boundary.75fac461.js | 1.12kb | +0.00b | 50.59s | -3.03s 🚀 |
| dist/media-picker-analytics-error-boundary.cfe02cc6.js | 966.00b | +0.00b | 50.59s | -3.03s 🚀 |
| dist/media-viewer-analytics-error-boundary.5e13c40c.js | 966.00b | +0.00b | 51.59s | -2.63s 🚀 |
| dist/media-card-analytics-error-boundary.b1a780f3.js | 962.00b | +0.00b | 51.55s | -2.64s 🚀 |
| dist/media-card-analytics-error-boundary.6f876fa2.js | 962.00b | +0.00b | 51.59s | -2.64s 🚀 |
| dist/simpleHasher.2a6e12b6.js | 643.00b | +0.00b | 50.60s | -3.02s 🚀 |
| dist/simpleHasher.85f7361e.js | 643.00b | +0.00b | 51.59s | -2.63s 🚀 |
| dist/index.html | 386.00b | +0.00b | 51.72s | -2.65s 🚀 |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.18s | -25.00ms |
| Cached | 391.00ms | -29.00ms 🚀 |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/309735411713835584)**

File: packages/transformers/image/src/ImageTransformer.js [hunk 1]
```diff
@@ -1,34 +1,66 @@
...
+      ? asset.query.as.toLowerCase().trim()
+      : null;
+    if (targetFormat && !FORMATS.has(targetFormat)) {
+      throw new Error(
+        `The image transformer does not support ${targetFormat} images.`,
+      );
+    }
+
+    const format = nullthrows(FORMATS.get(targetFormat || originalFormat));
+    const outputOptions = config[format];
```
Comment (devongovett) [inline packages/transformers/image/src/ImageTransformer.js:61]: Should we default `mozjpeg` to true? I think it offers better compression and I don't think there's a downside to it?

File: packages/transformers/image/src/ImageTransformer.js [hunk 2]
```diff
@@ -1,34 +1,66 @@
...
+  ['heif', 'heif'],
 ]);
 
 const SHARP_RANGE = '^0.28.3';
 
 export default (new Transformer({
-  async transform({asset, options}) {
+  async loadConfig({config}) {
+    let configFile: any = await config.getConfig(
+      ['sharp.config.json'], // '.sharprc', '.sharprc.json'
```
Comment (devongovett) [inline packages/transformers/image/src/ImageTransformer.js:24]: let's keep it simple, I think one option is fine :)

**Comment by devongovett** [comment]
This looks good. Should there be a way to define mappings between file types? Like, maybe I have a bunch of PNGs, and I want to convert them all to JPEGs or WebP. Right now you'd have to set the `as` query param on every image but maybe it should be possible to define this in the config file? Not sure...

**Review by devongovett — APPROVED** [review]
looks good. if you want to change the mozjpeg default in a separate pr that's fine

**Review by devongovett — APPROVED** [review]

**Pull request merged**